src/mongodb/scores.controller.ts

```diff
@@ -0,0 +1,370 @@
+import { Controller, Get, Post, Body, Patch, Param, Delete, HttpStatus, Res } from '@nestjs/common';
+import { ScoresService } from './scores.service';
+import { CreateScoreDto } from './dto/create-score.dto';
+import { UpdateScoreDto } from './dto/update-score.dto';
+import { FastifyReply } from 'fastify';
+
+@Controller('scores')
+export class ScoresController {
+  constructor(
+    private readonly scoresService: ScoresService
+  ) { }
+
+  @Post()
+  async create(@Res() response: FastifyReply, @Body() createScoreDto: any) {
+    try {
+
+      let confidence_scoresArr = [];
+      let anomaly_scoreArr = [];
+
+      let originalText = createScoreDto.original_text;
+      let responseText = createScoreDto.output[0].source;
+      let originalTextTokensArr = originalText.split("");
+      let responseTextTokensArr = responseText.split("");
+
+      let correctTokens = [];
+      let missingTokens = [];
+
+      let hindiVowelSignArr = ["ा", "ि", "ी", "ु", "ू", "ृ", "े", "ै", "ो", "ौ", "ं", "ः"];
+
+      let vowelSignArr = [];
+
+      let taVowelSignArr = [
+        "ா",
+        "ி",
+        "ீ",
+        "ு",
+        "ூ",
+        "ெ",
+        "ே",
+        "ை",
+        "ொ",
+        "ோ",
+        "ௌ",
+        "்",
+      ];
+
+      if (createScoreDto.language === "hi") {
+        vowelSignArr = hindiVowelSignArr;
+      } else if (createScoreDto.language === "ta") {
+        vowelSignArr = taVowelSignArr;
+      }
+
+      let tokenHexcodeData = this.scoresService.gethexcodeMapping(createScoreDto.language);
+      let tokenHexcodeDataArr = [];
+
+      await tokenHexcodeData.then((tokenHexcodedata: any) => {
+        tokenHexcodeDataArr = tokenHexcodedata;
+      });
+
+      let prevEle = '';
+      let isPrevVowel = false;
+
+
+      let originalTokenArr = [];
+      let responseTokenArr = [];
+
+
+      for (let originalTextELE of originalText.split("")) {
```

Comment:
Add Comments appropriately, a simple walkthrough of what we are doing with the code would help.